optimum/exporters/tflite/config.py

```diff
@@ -0,0 +1,33 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Common TensorFlow Lite configuration classes that handle most of the features for building model specific 
+configurations.
+"""
+
+from ...utils import DummyTextInputGenerator, logging
+from .base import TFLiteConfig
+
+
+logger = logging.get_logger(__name__)
+
+
+class TextEncoderTFliteConfig(TFLiteConfig):
+    """
+    Handles encoder-based text architectures.
+    """
+
+    DUMMY_INPUT_GENERATOR_CLASSES = (DummyTextInputGenerator,)
+    MANDATORY_AXES = ("batch_size", "sequence_length", ("multiple-choice", "num_choices"))
```

Comment:
What is the motivation for `MANDATORY_AXES` compared to the ONNX export?

Reply:
Mandatory axes are the axes which need to be specified to export to `TFLite` since only static shapes are supported (except the batch siize)

Reply:
Just ensuring we are aware:

https://www.tensorflow.org/api_docs/python/tf/lite/Interpreter#resize_tensor_input

Reply:
Yes, I tried that once. I do not think it always works. But I would suggest going that path for now and check for dynamic shapes later.

Reply:
+1 to that. Always better to do it progressively.